controller-server/src/test/java/com/yahoo/vespa/hosted/controller/maintenance/CloudEventReporterTest.java

```diff
@@ -0,0 +1,161 @@
+
+import com.yahoo.config.provision.HostName;
+import com.yahoo.config.provision.NodeType;
+import com.yahoo.config.provision.zone.ZoneId;
+import com.yahoo.vespa.hosted.controller.ControllerTester;
+import com.yahoo.vespa.hosted.controller.api.integration.aws.CloudEvent;
+import com.yahoo.vespa.hosted.controller.api.integration.aws.MockAwsEventFetcher;
+import com.yahoo.vespa.hosted.controller.api.integration.configserver.Node;
+import com.yahoo.vespa.hosted.controller.api.integration.organization.IssueId;
+import com.yahoo.vespa.hosted.controller.api.integration.organization.MockIssueHandler;
+import com.yahoo.vespa.hosted.controller.integration.ZoneApiMock;
+import com.yahoo.vespa.jdk8compat.Set;
+import org.junit.Test;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.*;
+
+/**
+ * @author olaa
+ */
+public class CloudEventReporterTest {
+
+    private ControllerTester tester = new ControllerTester();
+    private ZoneApiMock nonAwsZone = createZone("prod.zone3", "region-1", "other");
+    private ZoneApiMock awsZone1 = createZone("prod.zone1", "region-1", "aws");
+    private ZoneApiMock awsZone2 = createZone("prod.zone2", "region-2", "aws");
+
+
+    /**
+     * Test scenario:
+     * Consider three zones, two of which are based in AWS
+     * We want to test the following:
+     * 1. Non-AWS zone is completely ignored
+     * 2. Tenant hosts affected by cloud event are deprovisioned
+     * 3. Infrastructure hosts affected by cloud event are reported by IssueHandler
+     */
+    @Test
+    public void maintain() {
+        setUpZones();
+        CloudEventReporter cloudEventReporter = new CloudEventReporter(tester.controller(), Duration.ofMinutes(15), new JobControl(tester.curator()));
+
+        equalListsIgnoreOrder(List.of("host1.com", "host2.com", "host3.com"), getHostnames(nonAwsZone.getId()));
+        equalListsIgnoreOrder(List.of("host1.com", "host2.com", "host3.com"), getHostnames(awsZone1.getId()));
+        equalListsIgnoreOrder(List.of("host4.com", "host5.com", "confighost.com"), getHostnames(awsZone2.getId()));
+
+        mockEvents();
+        cloudEventReporter.maintain();
+
+        equalListsIgnoreOrder(List.of("host1.com", "host2.com", "host3.com"), getHostnames(nonAwsZone.getId()));
+        equalListsIgnoreOrder(List.of("host3.com"), getHostnames(awsZone1.getId()));
+        equalListsIgnoreOrder(List.of("host4.com", "confighost.com"), getHostnames(awsZone2.getId()));
+
+        Map<IssueId, MockIssueHandler.MockIssue> createdIssues = tester.serviceRegistry().issueHandler().issues();
+        assertEquals(1, createdIssues.size());
+        String description = createdIssues.get(IssueId.from("1")).issue().description();
+        assertEquals("[confighost]", description);
+
+    }
+
+    private void equalListsIgnoreOrder(List<?> l1, List<?> l2) {
```

Comment:
Consider returning `Set` from `getHostnames()` and use `assertEquals(Set.of(..), getHostnames(...));`